addons/mail/static/src/new/thread/message_service.js

```diff
@@ -198,51 +199,37 @@ export class MessageService {
     _update(message, data, fromFetch = false) {
         const {
             attachment_ids: attachments = message.attachments,
-            body = message.body,
             is_discussion: isDiscussion = message.isDiscussion,
             is_note: isNote = message.isNote,
             is_transient: isTransient = message.isTransient,
             linkPreviews = message.linkPreviews,
             message_type: type = message.type,
             model: resModel = message.resModel,
-            needaction_partner_ids = message.needaction_partner_ids,
             res_id: resId = message.resId,
-            subject = message.subject,
             subtype_description: subtypeDescription = message.subtypeDescription,
-            starred_partner_ids = message.starred_partner_ids,
-            trackingValues = message.trackingValues,
-            notifications = message.notifications,
             ...remainingData
         } = data;
-        for (const key in remainingData) {
-            message[key] = remainingData[key];
-        }
-        Object.assign(message, {
+        Object.assign(message, remainingData);
```

Comment:
Not related to PR but I have no idea what is `remainingData` and I don't understand why those are allowed to override if they contain `undefined`.

Reply:
Totally agree. This sucks. I choose to keep the same logic as before, while gaining less LOCs from introduction of `assignDefined`.

I think models / dataclasses should forbid assigning fields that are not declared.